test/extend_test.jl

```diff
@@ -1,23 +1,40 @@
 using HDF5
 using Test
 
-@testset "extendtest" begin
+@testset "extend" begin
+
+
+## mid-level routines
```

Comment:
These tests I think would make more sense in `test/dataspace.jl`. The `get_extent_dims` method is already tested pretty exhaustively there, so I think it just needs a new block of tests to run and check `set_extent_dims`.